### PR TITLE
Create Managed Network Powershell

### PR DESCRIPTION
**Pull request by yanfa317**
## Checklist

- [ ] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - Azure/azure-powershell-cmdlet-review-pr#369
    - [ ] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by markcowl** [comment]
@yanfa317 Please link to your cmdlet review in the PR comments

**Comment by yanfa317** [comment]
@markcowl  https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/369
Thanks!

**Comment by yanfa317** [comment]
@adxsdkps  We verified that this PR is from MNC team to create powershell for MNC

**Comment by yanfa317** [comment]
@markcowl Can you help us to check why the checks failed? It seems that it is because of "Set-Variable -Name ProgressPreference -Value 'SilentlyContinue';. D:\a\1\s\tools//GenerateHelp.ps1 -ValidateMarkdownHelp -GenerateMamlHelp -BuildConfig Debug" Thanks!

**Comment by yanfa317** [comment]
@markcowl  Can you help us review it? Thanks!

**Comment by yanfa317** [comment]
Hi, @wyunchi-ms Can you help us review this RP? And do you know why checks are keeping in state of "In process" for more than 24 hours? But in detail page of azure piplines, it seems that all of them have finished. Thanks!

**Comment by yanfa317** [comment]
@wyunchi-ms  @isra-fel Thanks so much! Can you help us to review this PR? We would like to close it as soon as possbile.  Thanks again!

File: src/ManagedNetwork/ManagedNetwork/StartupScripts/sample.ps1 [hunk 1]
```diff
@@ -0,0 +1 @@
+#Placeholder for future scripts: Please delete this file and uncomment <CopyToOutputDirectory>Always</CopytoOutputDirectory> within the <None Include="StartupScripts\*.ps1"> block in the .csproj file.
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/StartupScripts/sample.ps1]: What's this file for?

File: src/ManagedNetwork/ManagedNetwork/Az.ManagedNetwork.psd1 [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+# Generated on: 10/15/2019
+#
+
+@{
+
+# Script module or binary module file associated with this manifest.
+# RootModule = ''
+
+# Version number of this module.
+ModuleVersion = '0.0.1'
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Az.ManagedNetwork.psd1]: We suggest use `0.1.0` as the first version

File: src/ManagedNetwork/ManagedNetwork/Az.ManagedNetwork.psd1 [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+DotNetFrameworkVersion = '4.7.2'
+
+# Minimum version of the common language runtime (CLR) required by this module. This prerequisite is valid for the PowerShell Desktop edition only.
+# CLRVersion = ''
+
+# Processor architecture (None, X86, Amd64) required by this module
+# ProcessorArchitecture = ''
+
+# Modules that must be imported into the global environment prior to importing this module
+RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '1.4.0'; })
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Az.ManagedNetwork.psd1]: `1.6.2`

File: src/ManagedNetwork/ManagedNetwork/Az.ManagedNetwork.psd1 [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+# FormatsToProcess = @()
+
+# Modules to import as nested modules of the module specified in RootModule/ModuleToProcess
+NestedModules = @('.\Microsoft.Azure.PowerShell.Cmdlets.ManagedNetwork.dll')
+
+# Functions to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no functions to export.
+FunctionsToExport = @()
+
+# Cmdlets to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no cmdlets to export.
+CmdletsToExport = '*'
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Az.ManagedNetwork.psd1]: Do not use wildcards please

File: src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkGroup/UpdateAzManagedNetworkGroup.cs [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        public List<string> ManagementGroupIdList { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Azure ManagedNetwork subscription ids.")]
+        public List<string> SubscriptionIdList { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Azure ManagedNetwork virtual network ids.")]
+        public List<string> VirtualNetworkIdList { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Azure ManagedNetwork subnet ids.")]
+        public List<string> SubnetIdList { get; set; }
```
[the remark this answers is not in the document]
Comment (yanfa317) [reply src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkGroup/UpdateAzManagedNetworkGroup.cs]: May I ask why we don't want list here?

**Comment by yanfa317** [comment]
@isra-fel Thanks  for your comments! I have fixed the code based on them. There is one quesions: why do we want stirng [] instead of list? Also, please help us review it again. Thanks so much!

**Comment by isra-fel** [comment]
> @isra-fel Thanks for your comments! I have fixed the code based on them. There is one quesions: why do we want stirng [] instead of list? Also, please help us review it again. Thanks so much!

An array is eaiser to create and manipulate in powershell thant a list. It's a general rule in [our design guideline](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/parameter-best-practices.md#array-vs-enumerable-types) that every cmdlet should follow.

File: src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetwork/GetAzManagedNetwork.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+        /// <summary>
+        /// Gets or sets the ARM resource ID
+        /// </summary>
+        [Parameter(Mandatory = true, 
+            HelpMessage = HelpMessage.ResourceIdNameHelp,
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceIdCompleter("Microsoft.ManagedNetwork/managedNetworks")]
+        public string ResourceId { get; set; }
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetwork/GetAzManagedNetwork.cs]: `ValueFromPipelineByPropertyName = true` please

File: src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetwork/RemoveAzManagedNetwork.cs [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+        /// <summary>
+        /// Gets or sets the ARM resource ID
+        /// </summary>
+        [Parameter(Mandatory = true, 
+            HelpMessage = HelpMessage.ResourceIdNameHelp,
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceIdCompleter("Microsoft.ManagedNetwork/managedNetworks")]
+        public string ResourceId { get; set; }
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetwork/RemoveAzManagedNetwork.cs]: `ValueFromPipelineByPropertyName = true` please

File: src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkGroup/GetAzManagedNetworkGroup.cs [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+        /// <summary>
+        /// Gets or sets the ARM resource ID
+        /// </summary>
+        [Parameter(Mandatory = true, 
+            HelpMessage = HelpMessage.ResourceIdNameHelp,
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceIdCompleter("Microsoft.ManagedNetwork/managedNetworks/managednetworkgroups")]
+        public string ResourceId { get; set; }
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkGroup/GetAzManagedNetworkGroup.cs]: `ValueFromPipelineByPropertyName = true` please

File: src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkGroup/RemoveAzManagedNetworkGroup.cs [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+        /// <summary>
+        /// Gets or sets the ARM resource ID
+        /// </summary>
+        [Parameter(Mandatory = true, 
+            HelpMessage = HelpMessage.ResourceIdNameHelp,
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceIdCompleter("Microsoft.ManagedNetwork/managedNetworks/managednetworkgroups")]
+        public string ResourceId { get; set; }
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkGroup/RemoveAzManagedNetworkGroup.cs]: `ValueFromPipelineByPropertyName = true` please

File: src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkPeeringPolicy/GetAzManagedNetworkPeeringPolicy.cs [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+        /// <summary>
+        /// Gets or sets the ARM resource ID
+        /// </summary>
+        [Parameter(Mandatory = true, 
+            HelpMessage = HelpMessage.ResourceIdNameHelp,
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceIdCompleter("Microsoft.ManagedNetwork/managedNetworks/managednetworkpeeringpolicies")]
+        public string ResourceId { get; set; }
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkPeeringPolicy/GetAzManagedNetworkPeeringPolicy.cs]: `ValueFromPipelineByPropertyName = true` please

File: src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkPeeringPolicy/RemoveAzManagedNetworkPeeringPolicy.cs [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+        /// <summary>
+        /// Gets or sets the ARM resource ID
+        /// </summary>
+        [Parameter(Mandatory = true, 
+            HelpMessage = HelpMessage.ResourceIdNameHelp,
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceIdCompleter("Microsoft.ManagedNetwork/managedNetworks/managednetworkpeeringpolicies")]
+        public string ResourceId { get; set; }
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkPeeringPolicy/RemoveAzManagedNetworkPeeringPolicy.cs]: `ValueFromPipelineByPropertyName = true` please

Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/Commands/ManagedNetworkPeeringPolicy/RemoveAzManagedNetworkPeeringPolicy.cs]: `ValueFromPipelineByPropertyName = true` please

**Comment by isra-fel** [comment]
One moe thing -- ` Az.ManagedNetwork.md` needs to be regenerated, too. Please make sure all the placeholders (`{...}`) are replaced by either regenerating or hand-writing. Thanks

**Comment by yanfa317** [comment]
@isra-fel  Removed 3 files in tools. And delete online help link in Az.ManagedNetwork.md since we don't have help page yet

File: src/ManagedNetwork/ManagedNetwork/ChangeLog.md [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    * Overview of change #4
+        - Additional information about change #4
+
+    ## YYYY.MM.DD - Version X.Y.Z (Previous Release)
+    * Overview of change #1
+        - Additional information about change #1
+-->
+
+## Upcoming Release
+    * Create cmdlets for Az.ManagedNetwork.
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/ChangeLog.md]:
```suggestion
* Create cmdlets for Az.ManagedNetwork.
```
All the lines should indent 4 spaces less...

File: src/ManagedNetwork/ManagedNetwork/help/Az.ManagedNetwork.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+Locale: en-US
+---
+
+# Az.ManagedNetwork Module
+## Description
+{{ Module to create, get, remove and update managednetwork }}
+
+## Az.ManagedNetwork Cmdlets
+### [Get-AzManagedNetwork](Get-AzManagedNetwork.md)
+{{ Gets a managednetwork network. }}
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/help/Az.ManagedNetwork.md]: I mean you need to run help generation tool again to get rid of these placeholders {{}}

File: src/ManagedNetwork/ManagedNetwork/help/Az.ManagedNetwork.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
+---
+Module Name: Az.ManagedNetwork
+Module Guid: 00000000-0000-0000-0000-000000000000
+Help Version: 0.1.0.0
+Locale: en-US
+---
+
+# Az.ManagedNetwork Module
+## Description
+{{ Module to create, get, remove and update managednetwork }}
```
Comment (isra-fel) [inline src/ManagedNetwork/ManagedNetwork/help/Az.ManagedNetwork.md]: Except for this line -- it should be written by hand

**Pull request merged**